activities/ActivityTemplateVue.activity/js/components/journal.js

```diff
@@ -0,0 +1,54 @@
+var Journal = {
+  data: {
+		parent: null,
+	},
+  mounted() {
+		this.parent = this.$parent;
+    
+    let vm = this;
+    requirejs(["sugar-web/activity/activity", "sugar-web/env"], function (activity, env) {
+			env.getEnvironment(function (err, environment) {
+				
+				// Load context
+				if (environment.objectId) {
+					console.log("Existing instance");
+					activity.getDatastoreObject().loadAsText(function (error, metadata, data) {
+						if (error == null && data != null) {
+							let context = JSON.parse(data);
+							vm.parent.pawns = context.pawns;
```

Comment:
This code is specific it should not be there.

Reply:
Components are more generic now.